capsules/src/dals/README.md

```diff
@@ -0,0 +1,100 @@
+Dynamic App-Loading System (DALS)
+=================================
+
+## Description
+
+This document introduces the design for a novel system named **DALS** which allows 
+the Kernel to dynamically (at runtime) load and run apps.  The data for a new app 
+to be loaded and run by Tock is transmitted to a Tock-compatible board through any 
+standard  communication protocol (BLE, Zigbee, USB, etc.) which provides "acks" or a similar service
+to lower data rates.  This data is then pushed 
+segment-by-segment through **DALS** which places it in an empty space in flash space and
+then sets up the process-specific elements for the newly-loaded app.
+
+## Design
+
+The design of **DALS** consists of a series of capsules with one capsule, the **AppLoader**, considered
+to be trusted.  This main trusted capsule is the heart of **DALS** and contains the code which collects
+the raw app data from another capsule, moves it through a pipeline of steps, places the
+resulting app binary in an empty slot in flash space, and, finally, sets up its process-memory region
+so the scheduler can run it at the appropriate time.
+
+Basic Pipeline at the core of **DALS**.
+
+
+Collect raw data from peripheral capsule (BLE, Zigbee, etc.) -----> Decompress Data  ----->  Find empty slot in flash --- 
+
+---> Verify Data ------> Validate Data ----> Set up process-specific structures (stack, heap, etc.)
```

Comment:
Some of these happen in parallel right? Maybe there is a better way to illustrate that